cmd/xl-v1-object.go

```diff
@@ -730,49 +730,54 @@ func (xl xlObjects) deleteObject(bucket, object string) error {
 
 	isDir := hasSuffix(object, slashSeparator)
 
-	// If its a directory request, no need to read metadata.
 	if !isDir {
 		// Read metadata associated with the object from all disks.
 		metaArr, errs := readAllXLMetadata(xl.getDisks(), bucket, object)
-
 		// get Quorum for this object
 		_, writeQuorum, err = objectQuorumFromMeta(xl, metaArr, errs)
 		if err != nil {
 			return err
 		}
+		err = reduceWriteQuorumErrs(errs, objectOpIgnoredErrs, writeQuorum)
```

Comment:
This is little hard to read, the reason is that listObjectModtimes() requires the parameter 'errs' but we don't really need it.. we can consider 'nil' in metaArray as an error. I can fix it in another PR.